ragna/ui/main_page.py

```diff
@@ -0,0 +1,175 @@
+        )
+
+        self.left_sidebar = LeftSidebar(api_wrapper=self.api_wrapper)
+        self.left_sidebar.on_click_chat = lambda chat: self.on_click_chat(chat)
+        self.left_sidebar.on_click_new_chat = lambda event: self.open_modal()
+
+        self.right_sidebar = RightSidebar()
+
+        pn.state.location.sync(
+            self,
+            {"current_chat_id": "current_chat_id"},
+            on_error=lambda x: print(f"error sync on {x}"),
+        )
+
+        self.chats = []
+        self.refresh_data()
+
+    def refresh_data(self):
+        self.chats = self.api_wrapper.get_chats()
+        self.left_sidebar.chats = self.chats
+
+        if len(self.chats) > 0:
+            chat_id_exist = (
+                len([c["id"] for c in self.chats if c["id"] == self.current_chat_id])
+                > 0
+            )
+
+            if self.current_chat_id is None or not chat_id_exist:
+                self.current_chat_id = self.chats[0]["id"]
+
+            for c in self.chats:
+                if c["id"] == self.current_chat_id:
+                    self.central_view.set_current_chat(c)
+                    break
+
+    # Modal and callbacks
+    def open_modal(self):
+        self.modal = ModalConfiguration(
+            api_wrapper=self.api_wrapper,
+            new_chat_ready_callback=lambda new_chat_id: self.open_new_chat(new_chat_id),
+            cancel_button_callback=lambda event: self.on_click_cancel_button(event),
+        )
+
+        self.template.modal.objects[0].objects = [self.modal]
+        self.template.open_modal()
+
+    def open_welcome_modal(self, event):
+        self.modal = ModalWelcome(
+            close_button_callback=lambda: self.template.close_modal(),
+        )
+
+        self.template.modal.objects[0].objects = [self.modal]
+        self.template.open_modal()
+
+    def open_new_chat(self, new_chat_id):
+        # called after creating a new chat.
+        self.current_chat_id = new_chat_id
+        self.refresh_data()
+
+        self.template.close_modal()
+
+    def on_click_cancel_button(self, event):
+        self.template.close_modal()
+
+    # Left sidebar callbacks
+    def on_click_chat(self, chat):
+        self.central_view.set_loading(True)
+        self.current_chat_id = chat["id"]
+        self.central_view.set_current_chat(chat)
+        self.central_view.set_loading(False)
+
+    # Right sidebar callbacks
+    def show_right_sidebar(self, title, content):
+        self.right_sidebar.title = title
+        self.right_sidebar.content = content
+        self.right_sidebar.param.trigger("content")
+        self.right_sidebar.show()
+
+    @param.depends("current_chat_id", watch=True)
+    def update_subviews_current_chat_id(self, avoid_senders=[]):
+        try:
+            if self.left_sidebar is not None and self.left_sidebar not in avoid_senders:
+                self.left_sidebar.current_chat_id = self.current_chat_id
+        except Exception:
+            pass
```

Comment:
What could cause this exception and why can we ignore it?

Reply:
Like many other things you reported, that's a leftover of the dev process